scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -48,19 +51,42 @@
     (swap! map-atom dissoc key)
     (swap! map-atom assoc key value)))
 
-;; REVIEW: Review handle-status-update
+(defn container-status->failure-reason
+  [^V1PodStatus pod-status ^V1ContainerStatus status]
+  ; TODO map kubernetes failure reasons
+  (let [terminated (-> status .getState .getTerminated)]
+    (cond
```

Comment:
Use 'case'?

Reply:
Sometimes you want to look at the pod-status instead of the terminated state, so case won't work.

Reply:
Ok.